src/cpp/src/lm_encoding.cpp

```diff
@@ -126,7 +126,7 @@ std::pair<EncodedResults, int32_t> get_lm_encoded_results(
                                                 get_active_sequence_groups),
                                  active_sequence_groups.end());
 
-    while (active_sequence_groups.size() > 0) {
+    do {
         size_t total_num_tokens = 0;
 
         for (auto& sequence_group : active_sequence_groups) {
```

Comment:
BTW

@sbalandi should we use `active_sequence_groups` instead of  `sequence_groups` when compute `beam_offets` ?

Reply:
it looks like yes, we can use `active_sequence_groups` here, but that variant is ok too, as in non active num_running_seqs will be 0
I'll check it to be sure and fix in https://github.com/openvinotoolkit/openvino.genai/pull/1215